spin-observatory-plugin-deck/src/components/pipelines/PipelineExecutions.tsx

```diff
@@ -15,49 +18,96 @@ interface IPipelineExecutionsProps {
   appName: string;
   pipeline?: IPipeline;
   parameters: string[];
-  status: string[];
+  statuses: string[];
   dateRange: IDateRange;
+  onStatusChange: ({ statusCount }: { statusCount: any }) => void
 }
 
-export const PipelineExecutions = ({ appName, pipeline, parameters, status, dateRange }: IPipelineExecutionsProps) => {
+export const PipelineExecutions = ({ appName, pipeline, parameters, statuses, dateRange, onStatusChange }: IPipelineExecutionsProps) => {
   const [executions, setExecutions] = useState<IExecution[]>([]);
+  const [filteredExecutions, setFilteredExecutions] = useState<IExecution[]>([]);
+  const [statusCount, setStatusCount] = useState<any>({});
   const [isLoading, setIsLoading] = useState<boolean>(true);
   const styles = useStyles();
 
   useEffect(() => {
     if (!pipeline) {
       setExecutions([]);
+      setFilteredExecutions([]);
+      setStatusCount({});
       setIsLoading(false);
       return;
     }
 
     const requestParams = {
       pipelineName: pipeline.name,
       pageSize: REQUEST_PAGE_SIZE,
-      statuses: status,
       startDate: dateRange.start,
       endDate: dateRange.end,
     };
 
     getExecutions(appName, requestParams).then((resp) => {
       setExecutions(resp);
+      setFilteredExecutions(filterExecutions(resp));
+      setStatusCount(getStatusCount(resp));
       setIsLoading(false);
     });
   }, [pipeline]);
 
+  useEffect(() => {
+    onStatusChange(statusCount);
+  }, [statusCount]);
+
+  useEffect(() => {
+    setFilteredExecutions(filterExecutions(executions));
+  }, [statuses]);
+
   useInterval(async () => {
     if (!pipeline) return;
     const resp = await getExecutions(appName, {
       pipelineName: pipeline.name,
-      statuses: status,
       pageSize: REQUEST_PAGE_SIZE,
       startDate: dateRange.start,
       endDate: dateRange.end,
     });
+
     setExecutions(resp);
+    setFilteredExecutions(filterExecutions(resp));
+    setStatusCount(getStatusCount(resp));
     setIsLoading(false);
   }, POLL_DELAY_MS);
 
+  const filterExecutions = (ex: IExecution[]) => {
```

Comment:
I think it would be a little more readable and concise to use the array `filter` method here and have the callback check if the current execution's status is included in the array of selected statuses

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter